Objective-C/CBLDictionary.h

```diff
@@ -0,0 +1,60 @@
+//
+//  CBLDictionary.h
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 4/12/17.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "CBLReadOnlyDictionary.h"
+#import "CBLDictionaryFragment.h"
+@class CBLArray;
+@class CBLSubdocument;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** CBLDictionary protocol defines a set of methods for getting and setting dictionary data. */
+@protocol CBLDictionary <CBLReadOnlyDictionary, CBLDictionaryFragment>
+
+/** Set an object value by key. Setting the value to nil will remove the property. Allowed value 
+    types are NSArray, NSDate, NSDictionary, NSNumber, NSNull, NSString, CBLArray, CBLBlob,
+    CBLSubdocument. The NSArrays and NSDictionaries must contain only the above types. An NSDate 
+    object will be converted to an ISO-8601 format string. 
+    @param value    the object value.
+    @param key      the key. */
+- (void) setObject: (nullable id)value forKey: (NSString*)key;
+
+/** Set a dictionary as a content. Allowed value types are NSArray, NSDate, NSDictionary, NSNumber, 
+    NSNull, NSString, CBLArray, CBLBlob, CBLSubdocument. The NSArrays and NSDictionaries must 
+    contain only the above types.
+    @param dictionary  the dictionary. */
+- (void) setDictionary: (NSDictionary<NSString*,id>*)dictionary;
```

Comment:
This is different from setObjectForKey:. This is setting the content of the CBLDictionary with a platform dictionary (AKA JSON Dictionary).